test/functional/p2p_blockfilters.py

```diff
@@ -47,11 +47,25 @@ def set_test_params(self):
         self.rpc_timeout = 480
         self.num_nodes = 2
         self.extra_args = [
-            ["-blockfilterindex", "-peerblockfilters"],
-            ["-blockfilterindex"],
+            ["-blockfilterindex=basic", "-peerblockfilters"],
+            ["-blockfilterindex=basic", "-peerblockfilters=0"],
         ]
 
     def run_test(self):
+        self.stop_node(0)
+
+        self.log.info('Check that -peerblockfilters without -blockfilterindex raises init error')
+        err_msg = 'Error: Cannot set -peerblockfilters without -blockfilterindex.'
+        self.nodes[0].assert_start_raises_init_error(['-blockfilterindex=0', '-peerblockfilters'], err_msg)
+
+        self.log.info('Check that passing unknown -blockfilterindex type raises init error')
+        err_msg = 'Error: Unknown -blockfilterindex value foo.'
+        self.nodes[0].assert_start_raises_init_error(['-blockfilterindex=foo', '-peerblockfilters'], err_msg)
+
+        # Restart and restore default "chain" topology
+        self.start_node(0, extra_args=self.extra_args[0])
+        self.connect_nodes(1, 0)
```

Comment:
ISTM this stopping, restarting, and connecting can be avoided per https://github.com/bitcoin/bitcoin/pull/19720/files#r499226085.